### ISSUE TITLE
False positive for `return in a defer function has no effect`

### ISSUE THREAD

**Issue by connorszczepaniak-wk** [mgechev/revive#1528]
**Describe the bug**
Returning from a function literal inside of a defer triggers the defer linter falsely.

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with the following flags & configuration file:

```shell
revive -config=revive.toml ./...
```

```toml
[rule.defer]
  arguments=[["return"]]
```

With the following code:
```go
package main

func main() {}

func foo() {
	var fn func() int
	defer func() {
		fn = func() int {
			return 0
		}
	}()

	fn()
}

```

**Expected behavior**
This should not trigger the defer linter; returning within a function literal is contextually outside of the defer.

**Desktop (please complete the following information):**
 - OS: macOS (Sequoia 15.3.1)
 - Version of Go: 1.25.1

**Comment by chavacava** [comment]
Hi @connorszczepaniak-wk thanks for reporting the issue.

The PR #1531 should fix the problem. Could you please check if it works for your case?

**Comment by connorszczepaniak-wk** [comment]
Thanks, looks like it'll work for us! 👍
